chainntnfs/bitcoindnotify/bitcoind_test.go

```diff
@@ -243,32 +244,120 @@ func testHistoricalConfDetailsNoTxIndex(t *testing.T, rpcpolling bool) {
+	blockCache := blockcache.NewBlockCache(10000)
+
+	notifier := setUpNotifier(
+		t, bitcoindConn, hintCache, hintCache, blockCache,
+	)
+	defer notifier.Stop()
+
+	// Since the node has its txindex disabled, we fall back to scanning the
+	// chain manually. A outpoint unknown to the network should not be
+	// notified.
+	var unknownHash chainhash.Hash
+	copy(unknownHash[:], bytes.Repeat([]byte{0x10}, 32))
+	invalidOutpoint := wire.NewOutPoint(&unknownHash, 0)
+	unknownSpendReq, err := chainntnfs.NewSpendRequest(invalidOutpoint, testScript)
+	require.NoError(t, err, "unable to create spend request")
+	broadcastHeight := syncNotifierWithMiner(t, notifier, miner)
+	spendDetails, errSpend := notifier.historicalSpendDetails(
+		unknownSpendReq, uint32(broadcastHeight), uint32(broadcastHeight),
+	)
+	require.NoError(t, errSpend, "unable to retrieve historical spend details")
+	require.Equal(t, spendDetails, (*chainntnfs.SpendDetail)(nil))
+
+	// Now, we'll create a test transaction and attempt to retrieve its
+	// spending details. In order to fall back to manually scanning the
+	// chain, the transaction must be in the chain and not contain any
+	// unspent outputs. To ensure this, we'll create a transaction with only
+	// one output, which we will manually spend. The backend node's
+	// transaction index should also be disabled, which we've already
+	// ensured above.
+	outpoint, output, privKey := chainntnfs.CreateSpendableOutput(t, miner)
+	spendTx := chainntnfs.CreateSpendTx(t, outpoint, output, privKey)
+	const noOfBlocks = 100
+	spendTxHash, err := miner.Client.SendRawTransaction(spendTx, true)
+	require.NoError(t, err, "unable to broadcast tx")
+	broadcastHeight = syncNotifierWithMiner(t, notifier, miner)
+	err = chainntnfs.WaitForMempoolTx(miner, spendTxHash)
+	require.NoError(t, err, "tx not relayed to miner")
+	_, err = miner.Client.Generate(noOfBlocks)
+	require.NoError(t, err, "unable to generate blocks")
+
+	// Ensure the notifier and miner are synced to the same height to ensure
+	// we can find the transaction spend details when manually scanning the
+	// chain.
+	spendReq, err := chainntnfs.NewSpendRequest(outpoint, output.PkScript)
+	require.NoError(t, err, "unable to create conf request")
+	currentHeight := syncNotifierWithMiner(t, notifier, miner)
+	validSpendDetails, err := notifier.historicalSpendDetails(
+		spendReq, uint32(broadcastHeight), uint32(currentHeight),
+	)
+	require.NoError(t, err, "unable to retrieve historical spend details")
+
+	// Since the backend node's txindex is disabled and the transaction has
+	// confirmed, we should be able to find it by falling back to scanning
+	// the chain manually.
+	require.NotNil(t, validSpendDetails)
+	require.Equal(t, validSpendDetails.SpentOutPoint, outpoint)
+	require.Equal(t, validSpendDetails.SpenderTxHash, spendTxHash)
+	require.Equal(t, validSpendDetails.SpendingTx, spendTx)
+	require.Equal(t, validSpendDetails.SpendingHeight, int32(broadcastHeight+1))
```

Comment:
if indexing would be off this would have thrown error.

I compared `actualSpendHeight` with received BlockHeight from `validSpendDetails`.